easyagents/agents.py

```diff
@@ -60,12 +60,29 @@ def __init__(self,
         self._backend_agent: Optional[bcore._BackendAgent] = None
         return
 
-    def _prepare_callbacks(self, callbacks: List[core.AgentCallback]) -> List[core.AgentCallback]:
+    def _prepare_callbacks(self, callbacks: List[core.AgentCallback],
+                           default_plots: Optional[bool],
+                           default_plot_callbacks: List[plot._PlotCallback],
+                           ) -> List[core.AgentCallback]:
         """Adds the default callbacks and sorts all callbacks in the order
-        _PreProcessCallbacks, AgentCallbacks, _PostProcessCallbacks."""
+            _PreProcessCallbacks, AgentCallbacks, _PostProcessCallbacks.
+
+        Args:
+            callbacks: existing callbacks to prepare
+            default_plots: if set or if None and callbacks does not contain plots then the default plots are added
+            default_plot_callbacks: plot callbacks to add.
+        """
         preProcess = [plot._PreProcess()]
         agent = []
         postProcess = [plot._PostProcess()]
+
+        if default_plots is None:
```

Comment:
Wouldn't we have to check that no callbacks have been passed, additional to the is None check?

May understanding of semantics:
- False: do not use defaults
- True: use defaults
- None: use defaults when no callbacks have been passed at all

Reply:
my understanding:

    False: do not use defaults
    True: use defaults
    None: use defaults when no *plot* callbacks have been passed 
=> we get plot callbacks if only "duration" or "log" callbacks have beend passed.

Reply:
=> default_plots becomes false in line 82 if we find an instance of a plot callback